File: Assembly-FSharp/Assembly-FSharp/Commons/Utils/Vector3Utils.fs
namespace RageCure.Commons.Utils

module Vector3Utils =
    open UnityEngine

    let normalizeIfNot (vector: Vector3) =
        if vector.magnitude > 1.f then
            vector.normalized
        else
            vector

    let normalize (vector: Vector3) =
        vector.normalized

    let projectOnPlane (normal: Vector3) (vector: Vector3) =
        Vector3.ProjectOnPlane (vector, normal)

    let isNear (pos1: Vector3) (pos2: Vector3) =
        let ratioX = pos1.x - pos2.x
        let ratioY = pos1.y - pos2.y
        let ratioZ = pos1.z - pos2.z
        (ratioX <= 0.2f && ratioY <= 0.2f && ratioZ <= 0.2f)
    

File: Assets/_Scripts/Commons/Utils/Vector3Utils.cs
using UnityEngine;

namespace Styx.Commons.Utils
{
    public class Vector3Utils
    {
        public static Vector3 NormalizeIfNot (Vector3 vector)
        {
            if (vector.magnitude > 1f || vector.magnitude < 1f)
                return vector.normalized;
            return vector;
        }

        public static Vector3 NormalizeIfGreater (Vector3 vector)
        {
            if (vector.magnitude > 1f)
                return vector.normalized;
            return vector;
        }
    }
}
